components/widgets/SunriseSunsetWidget.tsx

```diff
@@ -24,12 +26,15 @@ const SunriseSunsetWidget: React.FC<SunriseSunsetWidgetProps> = ({
     return () => clearInterval(intervalId);
   }, []);
 
+  const adjustedSunrise = new Date(sunrise * 1000 + timezone * 1000);
+  const adjustedSunset = new Date(sunset * 1000 + timezone * 1000);
```

Comment:
Wciąż ta logika jest w zduplikowana (to samo jest w komponencie HourlyForecast). Myślę, że najlepiej by było gdzieś blisko miejsca gdzie pobierasz te dane przerobić je tak, żeby zamiast (niewygodnych) timestampów i timezone offsetu, operować na (wygodnych) obiektach `Date` uwzględniających ewentualne korekty związane ze strefami czasowymi

Reply:
dodalem przerabianie danych w useWeatherData